### PR TITLE
Displaying teams in the loading

### PR DESCRIPTION
**Pull request by GodFuper**
Without further ado, just a picture
<a href="https://ibb.co/7gkc2N0">image</a>

Optionally, can make display the overall rating, balance percentage, and so on.
Ideally, so that someone would add localization.

### CODE REVIEW

**Comment by Uveso** [comment]
What a nice feature!

Works really nice. (if we use Teams the right way)

I only want to show some strange behaviors i found:

I formed 6 teams with 1 AI in it, all other 10 AIs don't had a team:
(not a hypotetical setting, i use this sometimes)

on a low resulution:
LoadingTeam1

1920*1080:
LoadingTeam2

And 2 AIs per Team, 4 AIs without team:
LoadingTeam3

Also Map information would be nice, like map name, size, mex count etc.

**Comment by GodFuper** [comment]
I marked all the positions of the texts by eye. Yes, if you have a low resolution, then you need to come up with a different algorithm for drawing a large number of commands.

I didn't say it was finished code.
For the most part, this is just a concept, and I hope that other coders will also take the initiative and help complement the visual part.

**Comment by Uveso** [comment]
I see, so this PR is a proof of concept, will not further developed and you need/want help.

File: lua/ui/game/gamemain.lua [hunk 1]
```diff
@@ -320,6 +326,157 @@ local function LoadDialog(parent)
...
+				teamText:SetText('FFA')
+			end
+			topicCreating = false
+			LayoutHelpers.SetDimensions(dialogContent, width, (table.getsize(allPlayers) + 1) * 22)
+		end
+		
+		local n = 1
+		for k, player in players do
+			local playerText = UIUtil.CreateText(infoDialog, '', 15, UIUtil.bodyFont)
+			local playerColor = gameColors.PlayerColors[player.PlayerColor]
```
Comment (Uveso) [inline lua/ui/game/gamemain.lua:458]: `WARNING: Error running lua script: ...ogramdata\faforever\repo\fa\lua\ui\game\gamemain.lua(458): access to nonexistent global variable "gameColors"`

gameColors is not defined in gamemain.lua

File: lua/ui/game/gamemain.lua [hunk 2]
```diff
@@ -320,6 +326,157 @@ local function LoadDialog(parent)
...
+        local background = UIUtil.CreateNinePatchStd(self, '/scx_menu/lan-game-lobby/dialog/background/')
+
+        LayoutHelpers.FillParentFixedBorder(background, content, 64)
+
+        LayoutHelpers.DepthUnderParent(background, content)
+    end
+}
+
+function CreateViewMap(gameInfo, parent)
+    local scenarioInfo = MapUtil.LoadScenario(gameInfo.GameOptions.ScenarioFile)
```
Comment (Uveso) [inline lua/ui/game/gamemain.lua:363]: WARNING: Error running lua script: ...ogramdata\faforever\repo\fa\lua\ui\game\gamemain.lua(364): access to nonexistent global variable "MapUtil"

local MapUtil is missing

File: lua/ui/game/gamemain.lua [hunk 3]
```diff
@@ -320,6 +326,157 @@ local function LoadDialog(parent)
...
+    local MAP_PREVIEW_SIZE = parent.Width() / 3.5
+    local HYDROCARBON_ICON_SIZE = 10
+    local MASS_ICON_SIZE = 8
+
+    local dialogContent = Group(parent)
+    LayoutHelpers.SetDimensions(dialogContent, MAP_PREVIEW_SIZE + 5, MAP_PREVIEW_SIZE + 5)
+
+    local mapDialog = InfoDialog(parent, dialogContent)
+
+    local mapPreview = ResourceMapPreview(dialogContent, MAP_PREVIEW_SIZE, MASS_ICON_SIZE, HYDROCARBON_ICON_SIZE, true)
```
Comment (Uveso) [inline lua/ui/game/gamemain.lua:377]: WARNING: Error running lua script: ...ogramdata\faforever\repo\fa\lua\ui\game\gamemain.lua(380): access to nonexistent global variable "ResourceMapPreview"

local ResourceMapPreview is missing

**Comment by Uveso** [comment]
Looks already awesome.

Love the map preview

**Comment by Uveso** [comment]
Yes, works now. 

Can you do me a favor ?
You editor is using TAB instead of 4 spaces, can you change that ?
(All lines inserted by you have TABs at the beginning)

tabs

**Comment by GodFuper** [comment]
I still have a couple of ideas. In the near future, I will make it so that the name and dimensions of the map are displayed. I also think that need to display either the faction icon or the name of the faction that the player has chosen. When everything is ready, I will make the correct tabulation.

**Comment by Uveso** [comment]
@GodFuper 

Yes, the faction icon would be a nice. Looks more professional :)

Also a nice info would be a mass spot count, maybe divided by player.
Like `map has x mass spots, so every player has x/player mass spots available`

Changing the tabs at the end is fine; just do it in a separate commit, please.
Name it "cleanup" or something like that.

Can't wait to see your next update.

**Comment by GodFuper** [comment]
I was a little hasty and slightly combined "cleanup" and my code update. I wanted to finish it quickly.
Accidentally performed a little extra in lobby, which you yourself have already changed Over so.

I think this is the final build. Ideally, somebody still need to make a transfer to options, but there's not much I can do to help.. I'm not good with translations.
I thought that even more information would be superfluous.

**Comment by Garanas** [comment]
Closing this due to inactivity. We'll use the setup for something else however 😃 .

**Pull request closed without merging**